pymc3/distributions/multivariate.py

```diff
@@ -514,7 +493,11 @@ def random(self, point=None, size=None):
         array
         """
         a = draw_values([self.a], point=point, size=size)[0]
-        samples = generate_samples(self._random, a=a, dist_shape=self.shape, size=size)
+        output_shape = to_tuple(size) + a.shape
```

Comment:
Interesting. We were all time testing with parameter `a` being numpy arrays. 
In case, parameter `a` is PyMC3 distribution, then using `draw_values` function on it will return (size, dist_shape) shaped array. So, output shape will become `size + size + dist_shape`.

We can rely on `self.shape` here instead of `a.shape` to get concrete distribution shape - 
```suggestion
        output_shape = to_tuple(size) + to_tuple(self.shape)
```